src/sonarr.ts

```diff
@@ -1,136 +1,151 @@
 import { sonarrApi } from './fetch.ts';
 import { cleanTitle, hasExclusionTag } from './utils.ts';
 
+type SonarrApiClient = NonNullable<typeof sonarrApi>;
+
+interface SonarrSeries {
+  id: number;
+  title: string;
+  tvdbId: number;
+  tags: number[];
+}
+
+interface SonarrEpisode {
+  id: number;
+  seriesId: number;
+  tvdbId: number;
+  seasonNumber: number;
+  episodeNumber: number;
+  monitored: boolean;
+}
+
+interface SonarrTag {
+  id: number;
+  label: string;
+}
+
+async function getSonarrSeriesList(
+  api: SonarrApiClient,
+): Promise<SonarrSeries[] | null> {
+  const { data: seriesList, error } = await api.GET('/api/v3/series');
+  if (error) {
+    console.error('Failed to get series list from Sonarr:', error);
+    return null;
+  }
+  return seriesList as SonarrSeries[];
```

Comment:
Can we remove the type assertions throughout? They shouldn't be necessary if we're handling things properly